plans/create.pp

```diff
@@ -74,10 +80,38 @@
   Optional[String[1]] $environment               = 'production',
   Optional[Enum['linux', 'windows']] $os_type    = 'linux',
   Optional[Provision::ProviderOptions] $provider_options = undef,
+
+  ##GCP Specific Parameters
+  Optional[String[1]] $project                    = undef,
+  Optional[String[1]] $network                    = 'default',
+  Optional[String[1]] $subnetwork                 = 'default',
+  Optional[String[1]] $subnetwork_project         = undef,
 ) {
   out::message('Starting infrastructure provisioning')
   $tf_dir = "terraform/${provider}/"
 
+  # Validate the provider options for GCP
```

Comment:
I think this can be moved to `plans/terraform/apply.pp` ?
We can keep this plan clean and simple and add these complexities into nested one.

Reply:
What is the relation between `subnet` and `subnetwork`? We need to think of a minimal set of generic top-level params enabling us to create a gcp instance before needing to specify provider-specific params.

Reply:
Moved this part to a function. @Ramesh7 
Also @timidri , AFAIK, subnet in AWS is same as subnetwork in GCP. As per the suggestions of @Ramesh7 , we can move out network, subnetwork, subnetwork_project to provider_specific hash as well.